assets/forms/camera_settings.ui

```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" encoding="UTF-8"?>

```

Comment:
![image](https://user-images.githubusercontent.com/2094779/82744353-32466580-9d45-11ea-8486-10a1ee235c92.png)

Reply:
I know y'all don't prefer the visual layouts but the code to set up this layout is hella complicated.

Reply:
i feel like this is just harder to follow :p

i would prefer to read the code sequentially and see how it all gets defined

Reply:
This isn't just a code abstraction. It also uses the AUTOUIC module to give you access to all of the UI elements automatically and wire up the slots and signals with no code. I can tell you're not buying it. :b

Reply:
as strange as it may sound, i actually want to use Qt-stuff as little as possible 😬 

the more code I can see and follow the better hah

Reply:
I generally have the same philosophy as you but boilerplate code is boring. I generally prefer separating different tasks, even if all done in code. So layout would be one task. Communication between modules is another. Initializing the state is another. Etc.

The other advantage I find isn't as obvious yet but when you're playing around with the interface to find the best look it's easier to play around in the editor.

Reply:
i guess i have a different idea on organizing things

I have more of a hybrid oop/procedural where I group "logical units" into functions

so in QT-ness, think of this as every UI element exists in some function. When it comes to boilerplate stuff, i usually minimize that abstracting away the generic stuff, but i guess i didn't see the layout stuff as much of a nuisance